pkg/test/framework/components/cluster/factory.go

```diff
@@ -0,0 +1,41 @@
+//  Copyright Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package cluster
+
+import (
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+type Kind string
+
+const (
+	Kubernetes Kind = "Kubernetes"
+	Aggregate  Kind = "Aggregate"
+)
+
+type Config struct {
+	Kind                    Kind              `json:"kind,omitempty"`
+	Name                    string            `json:"name,omitempty"`
+	Network                 string            `json:"network,omitempty"`
+	ControlPlaneClusterName string            `json:"controlPlaneClusterName,omitempty"`
```

Comment:
In the [Deployment Models](https://istio.io/latest/docs/ops/deployment/deployment-models/) doc we define Primary as: 
> A cluster like this one, with its own local control plane, is referred to as a primary cluster.

The cluster could have the `ControlPlaneCluster` point to an external control plane, but have the `ConfigCluster` set to itself, or the remote config cluster.

Reply:
`Primary` just means that it's a cluster that runs a control plane, it doesn't really indicate anything about the source of config. If our docs say otherwise, it's merely due to the fact that we haven't officially introduced the concept of `config cluster` yet.

So with that said, conceptually `primary cluster` and `control plane cluster` are identical. Config cluster is just where a primary gets its config from.